src/flow-util.c

```diff
@@ -204,21 +204,26 @@ void FlowInit(Flow *f, const Packet *p)
     SCReturn;
 }
 
-int g_bypass_counter_id = -1;
+int g_bypass_info_id = -1;
 
-int GetFlowBypassCounterID(void)
+int GetFlowBypassInfoID(void)
 {
-    return g_bypass_counter_id;
+    return g_bypass_info_id;
 }
 
-static void FlowCounterFree(void *x)
+static void FlowBypassFree(void *x)
 {
+    FlowBypassInfo *fb = (FlowBypassInfo *) x;
+
+    if (fb->bypass_data && fb->BypassFree) {
```

Comment:
check below 'x' can be NULL, but dereference here in unconditional

Reply:
indeed